packages/react-server/src/ReactServerStreamConfigNode.js

```diff
@@ -33,11 +34,16 @@ export function flushBuffered(destination: Destination) {
   }
 }
 
-export function beginWriting(destination: Destination) {
-  // Older Node streams like http.createServer don't have this.
-  if (typeof destination.cork === 'function') {
-    destination.cork();
+export function beginWriting(destination: Destination): boolean {
+  // Older versions of Node don't have this.
+  if (destination.writableNeedDrain !== true) {
```

Comment:
How about `writableLength >= writableHighWaterMark`? It goes back a little further in terms of Node versions. That's what `write()` practically returns, and what is checked before needDrain is set to true. It's also how we implement that for web streams.